### DISCUSSION TITLE
Propose and implement an escape sequence for draggable files

### DISCUSSION THREAD

**Discussion by valpackett** [wezterm/wezterm#1135] [Ideas]
Imagine if we could just drag files out of `ls` output or a file manager like `ranger` and drop them into other applications…

That actually doesn't seem hard to implement. All that requires is a sequence similar to the "current directory" one, but with an end block (so it would apply to a particular string between start and end sequences). Programs like `exa`, `lsd`, `ranger` would be modified to emit those sequences. Then in the terminal just detect drag out of those marked cells and do a d&d offer with the provided path.

I might prototype this sometime in the future :)

**Comment by wez** [comment]
FWIW, I think the OSC 8 hyperlink sequence is probably sufficient for this, in terms of escape sequences, and programs are already supporting it (eg: `ls --hyperlink` in reasonably recent GNU `ls`) for clicking and opening `file://` URIs.

I think it would be a logical extension to allow dragging a cell with a hyperlink.  Some folks have already requested drag and drop in the opposite direction in https://github.com/wez/wezterm/issues/640
